File: src/C++/Autor/Origen/Biblioteca.cpp
#include "Biblioteca.h"
using namespace std;

Autor::Autor(string nombre, string nacionalidad) {
    this->nombre = nombre;
    this->nacionalidad = nacionalidad;
}

string Autor::toString() {
    return "Nombre: " + this->nombre + "\nNacionalidad: " + this->nacionalidad;
}

Libro::Libro(string titulo, int anioPublicacion, Autor a): autor(a) {
    this->titulo = titulo;
    this->anioPublicacion = anioPublicacion;
    // this->autor = autor;
}

string Libro::ToString() {
    return this->autor.toString() + "\nTitulo: " + this->titulo + "\nAnio de Publicacion: " + to_string(this->anioPublicacion);
}

File: src/Python/Parcial1/Autor/biblioteca.py
class Autor:
    def __init__(self, nombre, nacionalidad):
        self.nombre = nombre
        self. nacionalidad = nacionalidad

    def __str__(self): # Equivale al método toString()
        return f"Autor: {self.nombre}\nNacionalidad: {self.nacionalidad}"

class Libro:
    def __init__(self, titulo, anioPublicacion, autor):
        self.titulo = titulo
        self.anioPublicacion = anioPublicacion
        self.autor = autor # Objeto de la clase Autor como atributo

    def __str__(self):
        return f"{self.autor}\nTitulo: {self.titulo}\nAnio Publicacion: {self.anioPublicacion}" 
    